### PR TITLE
nav: add arrow images for RHD instructions

### PR DESCRIPTION
**Pull request by eFiniLan**
Navigation instruction icon on RHD is not handled:
DSC_0711

This will show the right icon instead:
DSC_0712

### CODE REVIEW

File: selfdrive/ui/qt/maps/map.cc [hunk 1]
```diff
@@ -319,6 +320,7 @@ void MapWindow::offroadTransition(bool offroad) {
 }
 
 MapInstructions::MapInstructions(QWidget * parent) : QWidget(parent) {
+  is_rhd = Params().getBool("IsRHD");
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/maps/map.cc:323]: can we get this from mapbox instead?

Comment (eFiniLan) [reply selfdrive/ui/qt/maps/map.cc:323]: Hi @adeebshihadeh,
thanks! according to the [direction api](https://docs.mapbox.com/api/navigation/directions/):

`driving_side` column is `The legal driving side at the location for this step. Either left or right.`

however, it is under `routes[0] > legs[0] > steps[0] > driving_side`
 
image

any idea where I will have access to that? looks like `cur_maneuver.extendedAttributes()` does not have it.

Comment (pd0wm) [reply selfdrive/ui/qt/maps/map.cc:323]: The parser from the qtlocation plugin is not very complete. It's on my own list to write my own parser so we can also get speed limits properly.

**Comment by pd0wm** [comment]
Did you make the icons yourself? I got the other ones from https://github.com/mapbox/directions-icons/tree/master/src/png/light and didn't include rhd versions.

File: selfdrive/ui/qt/maps/map.cc (hunk 1 above)
Comment (pd0wm) [reply selfdrive/ui/qt/maps/map.cc:323]: So we can use the RHD toggle/param for now and switch to this output once we have our own parser.

**Comment by eFiniLan** [comment]
> Did you make the icons yourself? I got the other ones from https://github.com/mapbox/directions-icons/tree/master/src/png/light and didn't include rhd versions.

yeah... I just flip them : )

File: selfdrive/ui/qt/maps/map.cc [hunk 2]
```diff
@@ -470,8 +478,15 @@ void MapInstructions::updateInstructions(cereal::NavInstruction::Reader instruct
     } else if (straight) {
       fn += "turn_straight";
     }
+    fn +=  + ".png";
```
Comment (pd0wm) [inline selfdrive/ui/qt/maps/map.cc:481]: I assume the lane arrows don't need a RHD version?

Comment (eFiniLan) [reply selfdrive/ui/qt/maps/map.cc]: this section do not append modifier, I've removed them :+1:

**Comment by pd0wm** [comment]
What do you think? No need for string replacements this way.

**Comment by pd0wm** [comment]
2021-11-12-141437_1122x760_scrot

What program did you use to flip the images? They are larger in size, but lower in quality.

**Comment by eFiniLan** [comment]
> 2021-11-12-141437_1122x760_scrot
> 
> What program did you use to flip the images? They are larger in size, but lower in quality.
> 
> Maybe it's better to let Qt deal with the flipping.

This was done with gimp, I just tried macos "Preview" and photoshop, photoshop version looks better and smaller.

flipping could work too, but then we will need to specify *roundabout*/*rotary* and *uturn* types so I guess using images is cleaner.

**Comment by pd0wm** [comment]
Still seeing some more aliasing on the flipped ones.

We don't need to detect roundabout/rotary, right? We can just swap left/right for every instruction and then flip the image. For a lot instruction this won't do anything, but that doesn't really matter.

**Comment by eFiniLan** [comment]
> Still seeing some more aliasing on the flipped ones.
> 
> We don't need to detect roundabout/rotary, right? We can just swap left/right for every instruction and then flip the image. For a lot instruction this won't do anything, but that doesn't really matter.

ahh that will do, I'll give it a shot :)

**Comment by pd0wm** [comment]
Much more elegant this way!

**Pull request merged**